### PR TITLE
MODULES-5135 - Rake task to execute string wrapper

### PR DESCRIPTION
**Pull request by pmcmaw**
This is work for i18N, this is a rake task that executes the string
wrapper script for modules.

### CODE REVIEW

File: lib/string_prep/string_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Run with `ruby-rewrite -m --load string_wrapper.rb <file_or_dir_to_mark>`. This will modify the file(s) in place.
+#
+# Note: the changes made by this script still need a thorough manual review, to ensure that _only_ strings that need to be externalized are marked. More sophisticated matching criteria will be added as they are discovered through use of the script on complex files.
+#
+# For more info on the parser/rewriter, see <https://github.com/whitequark/parser>.
+#
+# This script was originally written by Maggie Dryer and changes made by Eric Putnam.
+
+require 'pry'
+require 'pry-nav'
```
Comment (DavidS) [inline lib/string_prep/string_wrapper.rb]: The script below doesn't seem to use pry. Please remove debugging tools before committing!

File: lib/string_prep/string_wrapper.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      # ignore strings with no whitespace - are typically keywords or interpolation statements and cover the above commented-out statements
+      return if str =~ /^\S*$/
+      insert_before(node.loc.begin, '_(')
+      insert_after(node.loc.end, ')')
+    end
+  end
+
+  def on_send(node)
+    method_name = node.loc.selector.source
+    return unless /call_function|function_deprecation|new(param|function|property)|puts|raise|desc|\+|.*[Ee]rror|[Dd]ebug|warning|warn|fail|notice/.match(method_name)
```
Comment (eputnam) [inline lib/string_prep/string_wrapper.rb]: we need to talk about what we want to start decorating. and/or if we want to be able to pass values like `error` or `warning` to the rake task.

File: lib/string_prep/test.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+something = 'something'
+
+case testvar
+when 'onething'
+  function
+  raise Puppet::Error _('Something went wrong here')
+when 'anotherthing'
+  function2
+end
```
Comment (eputnam) [inline lib/string_prep/test.rb]: do we need to include this file?

**Comment by HelenCampbell** [comment]
@smcclellan Hey, would you be able to review this for us please when you get a chance? Thanks!

File: lib/tasks/gettext.rake [hunk 4]
```diff
@@ -45,6 +45,14 @@ namespace :gettext do
     end
   end
 
+  desc 'Run string_wrapper.rb against all Ruby files within a module'
+  task :wrap_ruby_string do
+    begin
+      script_dir = File.expand_path('../string_prep/string_wrapper.rb', File.dirname(__FILE__))
```
Comment (er0ck) [inline lib/tasks/gettext.rake]: will this ever be used in windows?

**Comment by pmcmaw** [comment]
There has been a change of direction on this piece of work. Closing.

**Pull request closed without merging**